File: client/js/messages/LoginMessage.js
class LoginMessage extends Message {
    constructor (user, pass) {
        super();
        this.user = user;
        this.pass = pass;
    }
    
    accept (visitor) {
        visitor.visitLogin (this);
    }
    
    toString () {
        return "LOGIN_R " + this.user + " " + this.pass;
    }
}

File: server/messages/LoginMessage.cpp
#include "LoginMessage.h"
#include "MessageVisitor.h"

LoginMessage::LoginMessage (const std::string& user, const std::string& pass): _user(user), _pass(pass) 
{

}

std::string LoginMessage::accept (MessageVisitor* visitor, Session* session) 
{
    return visitor->visitLogin(this, session);
}
